ui/app/routes/vault/cluster/secrets/backend/configuration/edit.ts

```diff
@@ -47,10 +48,21 @@ export default class SecretsBackendConfigurationEdit extends Route {
       // ex: adapterPath = ssh/ca-config, convert to: ssh-ca-config so that you can pass to component @model={{this.model.ssh-ca-config}}
       const standardizedKey = adapterPath.replace(/\//g, '-');
       try {
-        model[standardizedKey] = await this.store.queryRecord(adapterPath, {
+        const configModel = await this.store.queryRecord(adapterPath, {
           backend,
           type,
         });
+        // some of the models return a 200 if they are not configured (ex: azure)
+        // so instead of checking a catch or httpStatus, we check if the model is configured based on the getter `isConfigured` on the engine's model
+        // if the engine is not configured we update the record to get the default values
```

Comment:
Azure, at the moment, is the only engine that returns a 200 when the config has not been modified. However, future engines might also keep this pattern / or it could just be an Azure thing. For now, I have this line to handle it and `isConfigured` getter added to the model. Is it my favorite? No, but I feel like it's pretty smooth for the situation. 🤷‍♀️